src/Samples/SarifTrim/Program.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+namespace SarifTrim
+{
+    /// <summary>
+    ///  SarifTrim is a one-off sample removing some redundant and some excessively large parts of a Sarif log
+    ///  for handling of a huge log.
+    /// </summary>
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            if (args.Length < 1)
+            {
+                Console.WriteLine(@"Usage: SarifTrim [inputFilePath] [outputFilePath?] [removeParts?]
+    removeParts: Comma-separated sections to remove. Options: UriBaseIds,CodeFlows,RelatedLocations,Graphs,GraphTraversals,Stacks,WebRequests,WebResponses");
+                return;
+            }
+
+            string inputFilePath = args[0];
+            string outputFilePath = (args.Length > 1 ? args[1] : Path.ChangeExtension(inputFilePath, $"trim{Path.GetExtension(inputFilePath)}"));
+
+            // Split on comma, remove whitespace, and put in case-insensitive HashSet
+            HashSet<string> removeParts = new HashSet<string>((args.Length > 2 ? args[2] : "").Split(',').Select(entry => entry.Trim()), StringComparer.OrdinalIgnoreCase);
+
+            SarifLog log = null;
+            using (new ConsoleWatch($"Loading \"{inputFilePath}\"..."))
+            {
+                log = SarifLog.LoadDeferred(inputFilePath);
+            }
+
+            using (new ConsoleWatch($"Trimming \"{inputFilePath}\" into \"{outputFilePath}\"..."))
+            {
+                Run run = log.Runs[0];
+
+                SarifConsolidator consolidator = new SarifConsolidator(run);
+                consolidator.MessageLengthLimitBytes = 1024;
+                consolidator.RemoveCodeFlows = (removeParts.Contains("CodeFlows"));
+                consolidator.RemoveRelatedLocations = (removeParts.Contains("RelatedLocations"));
+                consolidator.RemoveGraphs = (removeParts.Contains("Graphs"));
+                consolidator.RemoveGraphTraversals = (removeParts.Contains("GraphTraversals"));
+                consolidator.RemoveStacks = (removeParts.Contains("Stacks"));
+                consolidator.RemoveWebRequests = (removeParts.Contains("WebRequests"));
+                consolidator.RemoveWebResponses = (removeParts.Contains("WebResponses"));
+
+                // Trim Result CodeFlowLocations, long Messages, and redundant Region properties
+                using (SarifLogger logger = new SarifLogger(outputFilePath, LoggingOptions.OverwriteExistingOutputFile, tool: run.Tool, run: run))
```

Comment:
It would be nice if the sample code used CommandlineParser and the built in options base classes, in order to help keep command line interfaces uniform and slow the necessity of authoring new, redundant logic to handle things like file overwrites. I can appreciate that there is another value in sample code, which is to keep dependencies minimal (to ease portability of the code). But if we will be using the samples as a kind of staging ground for things that move into the 'proper' SDK binaries, probably better to consume those existing patterns.

---
In reply to: [323979575](https://github.com/microsoft/sarif-sdk/pull/1672#discussion_r323979575) [](ancestors = 323979575)

Reply:
Since Mike wants me to move SarifTrim and SarifToCsv into the Multitool imminently, I'm going to leave these unchanged.